### ISSUE TITLE
Incompatibility with Gradle 8.3 when using Kotlin. "The value for property 'languageVersion' is final and cannot be changed any further."

### ISSUE THREAD

**Issue by Spikhalskiy** [micronaut-projects/micronaut-gradle-plugin#804]
### Expected Behavior

Upgrade from Gradle 8.2 to Gradle 8.3 doesn't cause build failure for an existing micronaut project.

### Actual Behaviour

Adding
```
    id("io.micronaut.application") version "4.0.2"
```
into `build.gradle.kts` of the standard Kotlin+Gradle application generated by Idea now breaks the build with

```
* What went wrong:
The value for property 'languageVersion' is final and cannot be changed any further.
```

on line 
```
kotlin {
    jvmToolchain(17)
}
```

### Steps To Reproduce

See an attached github repo with a sample project failing to build.
Alternatively, just create a project like it's described here: 
https://kotlinlang.org/docs/get-started-with-jvm-gradle-project.html#create-a-project
Add  `id("io.micronaut.application") version "4.0.2"` into `plugins` section and `micronaut { version = "4.0.2" }` block
Build works on Gradle 8.2 and 8.2.1
Upgrade Gradle to 8.3. Build fails with the mentioned error.

### Environment Information

JDK: 17
Kotlin 1.9.0
Micronaut Plugin 4.0.2

### Example Application

https://github.com/Spikhalskiy/gradle-8.3-micronaut-failure

### Version

Plugin 4.0.2

**Comment by rjaros** [comment]
Related issue: https://github.com/gradle/gradle/issues/26061

**Comment by Spikhalskiy** [comment]
The second solution `... use the attributeProvider to lazily query the attributes of the source Configuration (runtimeClasspath).` proposed by Gradle team to plugin authors could work nicely.

**Comment by Spikhalskiy** [comment]
The plugin that causes the issue is "io.micronaut.graalvm". 
I tried to replace id("io.micronaut.application") version "4.0.2" with
```
id("io.micronaut.minimal.application") version "4.0.2"
id("io.micronaut.docker") version "4.0.2"
id("io.micronaut.crac") version "4.0.2"
```
and it builds. Which can be a workaround for this issue for people who don't need graal.

There is also a good chance that it's not an issue with micronaut plugin at all and the actual issue is in Graal plugin:
https://github.com/graalvm/native-build-tools/issues/486

**Comment by melix** [comment]
This is fixed in release 4.1.0
